test/integration/etcd_storage_path_test.go

```diff
@@ -982,201 +561,381 @@ func TestEtcdStoragePath(t *testing.T) {
+	mediaType, _, err := mime.ParseMediaType(contentType)
+	if err != nil {
+		return nil, fmt.Errorf("the content type specified in the client configuration is not recognized: %v", err)
+	}
+	info, ok := runtime.SerializerInfoForMediaType(mediaTypes, mediaType)
+	if !ok {
+		if len(contentType) != 0 || len(mediaTypes) == 0 {
+			return nil, fmt.Errorf("no serializers registered for %s", contentType)
+		}
+		info = mediaTypes[0]
+	}
+
+	internalGV := unversioned.GroupVersions{
+		{
+			Group:   config.GroupVersion.Group,
+			Version: runtime.APIVersionInternal,
+		},
+		// always include the legacy group as a decoding target to handle non-error `Status` return types
+		{
+			Group:   "",
+			Version: runtime.APIVersionInternal,
+		},
+	}
+
+	s := &restclient.Serializers{
+		Encoder: config.NegotiatedSerializer.EncoderForVersion(info.Serializer, *config.GroupVersion),
+		Decoder: config.NegotiatedSerializer.DecoderToVersion(info.Serializer, internalGV),
+
+		RenegotiatedDecoder: func(contentType string, params map[string]string) (runtime.Decoder, error) {
+			info, ok := runtime.SerializerInfoForMediaType(mediaTypes, contentType)
+			if !ok {
+				return nil, fmt.Errorf("serializer for %s not registered", contentType)
+			}
+			return config.NegotiatedSerializer.DecoderToVersion(info.Serializer, internalGV), nil
+		},
+	}
+	if info.StreamSerializer != nil {
+		s.StreamingSerializer = info.StreamSerializer.Serializer
+		s.Framer = info.StreamSerializer.Framer
+	}
+
+	return s, nil
+}
+
+func isInCreateAndCompareWhiteList(obj runtime.Object) bool {
+	switch obj.(type) {
+	case *authorizationapiv1.ClusterPolicyBinding, *authorizationapiv1.ClusterPolicy: // TODO figure out how to not whitelist these
+		return true
+		// Removed this case per soltysh's request to not have so many exceptions, but leaving here so people are not confused by the errors
+		//case *apisbatchv2alpha1.CronJob: // since we do not cleanup once a test is failed, we will get an AlreadyExists error since ScheduledJob aliases CronJob
+		//	return true
+	}
+	return false
+}
+
+func isInInvalidNameWhiteList(obj runtime.Object) bool {
+	switch obj.(type) {
+	case *sdnapi.HostSubnet, *sdnapi.NetNamespace: // TODO figure out how to not whitelist these
+		return true
+	}
+	return false
+}
+
+func isInKindAndPathWhiteList(kind, pkgPath string) bool {
+	switch {
+	// aliases for templateapiv1.Template
+	case kind == "TemplateConfig" && pkgPath == "github.com/openshift/origin/pkg/template/api/v1",
+		kind == "ProcessedTemplate" && pkgPath == "github.com/openshift/origin/pkg/template/api/v1":
+		return true
+	}
+	return false
+}
+
+func getFromEtcd(keys etcd.KeysAPI, path string, gvk unversioned.GroupVersionKind) (runtime.Object, error) {
 	response, err := keys.Get(context.Background(), path, nil)
 	if err != nil {
 		return nil, err
 	}
-	output, _, err := d.Decode([]byte(response.Node.Value), gvk, nil)
+	output, err := jsonToObject(response.Node.Value, gvk)
```

Comment:
@liggitt I am starting to believe this change was incorrect.  I was trying to get away from our conversion code but I am thinking in this case I would want to use it to correctly covert the internal object stored in etcd to the correct external object.